src/Components/ServiceScene/JSONPanel/LabelRenderer.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+import { isNumber } from 'lodash';
+
+import { Field } from '@grafana/data';
+import { AdHocFiltersVariable, AdHocFilterWithLabels } from '@grafana/scenes';
+
+import { isLogLineField } from '../../../services/fields';
+import { jsonLabelWrapStyles } from '../../../services/JSONViz';
+import {
+  JsonDataFrameLabelsName,
+  JsonDataFrameStructuredMetadataName,
+  JsonDataFrameTimeName,
+  JsonVizRootName,
+  LabelsDisplayName,
+  LogsJsonScene,
+  NodeTypeLoc,
+  StructuredMetadataDisplayName,
+} from '../LogsJsonScene';
+import { KeyPath } from '@gtk-grafana/react-json-tree/dist/types';
+
+interface LabelRendererProps {
+  fieldsVar: AdHocFiltersVariable;
+  jsonFiltersSupported: boolean | undefined;
+  jsonParserPropsMap: Map<string, AdHocFilterWithLabels>;
+  keyPath: KeyPath;
+  lineField: Field;
+  lineFilters: AdHocFilterWithLabels[];
+  model: LogsJsonScene;
+  nodeType: string;
+}
+
+export default function LabelRenderer({
+  fieldsVar,
+  jsonFiltersSupported,
+  jsonParserPropsMap,
+  keyPath,
+  lineField,
+  lineFilters,
+  model,
+  nodeType,
+}: LabelRendererProps) {
+  const nodeTypeLoc = nodeType as NodeTypeLoc;
+  if (keyPath[0] === JsonDataFrameStructuredMetadataName) {
+    return <strong className={jsonLabelWrapStyles}>{StructuredMetadataDisplayName}</strong>;
+  }
+  if (keyPath[0] === JsonDataFrameLabelsName) {
+    return <strong className={jsonLabelWrapStyles}>{LabelsDisplayName}</strong>;
+  }
+
+  if (keyPath[0] === JsonVizRootName) {
+    return model.renderNestedNodeButtons(keyPath, jsonFiltersSupported);
+  }
+
+  // Value nodes
+  if (
+    nodeTypeLoc !== 'Object' &&
```

Comment:
This moved to a function would look beautiful. I found this unappealing on the first pass but failed to put words to it. No need to do it now, minor thing.